src/openms/include/OpenMS/QC/DBSuitability.h

```diff
@@ -0,0 +1,208 @@
+      Size num_top_novo = 0;
+
+      /// number of times the top hit is considered to be a database hit
+      Size num_top_db = 0;
+      
+      /// number of times a deNovo hit scored on top of a database hit
+      Size num_interest = 0;
+
+      /// number of times a deNovo hit scored on top of a database hit,
+      /// but their score difference was small enough, that it was still counted as a database hit
+      Size num_re_ranked = 0;
+
+      /// the cut-off that was used to determine when a score difference was "small enough"
+      /// this is normalized by mw
+      double cut_off = DBL_MAX;
+
+      /// the suitability of the database used for identification search, calculated with:
+      ///               #db_hits / (#db_hits + #deNovo_hit)
+      /// can reach from 0 -> the database was not at all suited to 1 -> the perfect database was used
+      ///
+      /// Preliminary tests have shown that databases of the right organism or close related organisms
+      /// score around 0.9 to 0.95, organisms from the same class can still score around 0.8, organisms
+      /// from the same phylum score around 0.5 to 0.6 and after that it quickly falls to suitabilities
+      /// of 0.15 or even 0.05.
+      /// Note that these test were only performed for one mzML and your results might differ.
+      double suitability = 0;
+    };
+
+    /// Constructor
+    /// Settings are initialized with their default values:
+    /// no_rerank = false, reranking_cutoff_percentile = 1, FDR = 0.01
+    DBSuitability();
+
+    /// Destructor
+    ~DBSuitability() = default;
+
+    /**
+    * @brief Computes suitability of a database used to search a mzML
+    *
+    * Counts top deNovo and top database hits. The ratio of db hits vs
+    * all hits yields the suitability.
+    * To re-rank cases, where a de novo peptide scores just higher than
+    * the database peptide, a decoy cut-off is calculated. This functionality
+    * can be turned off. This will result in an underestimated suitability,
+    * but it can solve problems like different search engines or to few decoy hits.
+    * 
+    * Parameters can be set using the functionality of DefaultParamHandler.
+    * Parameters are:
+    *           no_rerank                   - re-ranking can be turned off with this
+    *           reranking_cutoff_percentile - percentile that determines which cut-off will be returned
+    *           FDR                         - q-value that should be filtered for
+    *                                         Preliminary tests have shown that database suitability
+    *                                         is rather stable across common FDR thresholds from 0 - 5 %
+    *
+    * Since q-values need to be calculated the identifications are taken by copy.
+    *
+    * Result is appended to the result member. This allows for multiple usage.
+    *
+    * @param pep_ids      vector containing pepIDs with target/decoy annotation coming from a deNovo+database 
```

Comment:
add a comment why this is not a const& (`vector is modified internally, and is thus copied`) or similar